parser_library/src/processing/instruction_sets/instruction_processor.h

```diff
@@ -37,15 +37,15 @@ class instruction_processor : public diagnosable_ctx
     virtual void process(context::unique_stmt_ptr stmt) = 0;
     virtual void process(context::shared_stmt_ptr stmt) = 0;
 
-    virtual void collect_diags() const override {}
+    virtual void collect_diags() const override { collect_diags_from_child(eval_ctx); }
 
 protected:
     context::hlasm_context& hlasm_ctx;
     attribute_provider& attr_provider;
     branching_provider& branch_provider;
     workspaces::parse_lib_provider& lib_provider;
 
-    expressions::evaluation_context eval_ctx;
+    mutable expressions::evaluation_context eval_ctx;
```

Comment:
must this member variable even exist? can it be just created locally in `collect_diags`?

Reply:
the collecting of diagnostics in CA expressions has changed. It used to be that the expression object itself held the diagnostic object. Now, it is the evaluation context that collects diags (it implements `diagnosable_ctx`). Hence, it needs to live until the `instruction_processor::collect_diags` is called.